### PR TITLE
Added a Basic Genetic algorithm

### PR DESCRIPTION
**Pull request by Kaiyoto**
This Genetic algorithm recreates a target string by selecting random characters.
Efficiency: medium

### CODE REVIEW

**Comment by ParthS007** [comment]
This pull request **introduces 2 alerts** when merging 9a9986afc2bd2594e9988f6538e84e0725be0d0c into b63a1115b23fcd9e32db6f28d6a569b80252d18e - [view on LGTM.com](https://lgtm.com/projects/g/TheAlgorithms/Python/rev/pr-6d63365c1319ea57c8dcaefc784b5c53becc4f6d)

**new alerts:**

* 1 for Testing equality to None
* 1 for Result of integer division may be truncated

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: other/genetic_algorithm.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def __str__(self):
+        return 'String: ' + ''.join(self.data) + ', Score: ' + str(self.score)
+
+# SETUP
+for i in range(population):
+    randomString = ''
+    for j in range(len(target)):
+        randomString += random.choice(alphabet)      
+    data.append(Item(randomString, target))
```
Comment (jrmylow) [inline other/genetic_algorithm.py]: Maybe consider more compact code for generating random string:

`randomString = ''.join(random.choices(alphabet, k = len(target)))`

File: other/genetic_algorithm.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    sortedData = sorted(data, key=lambda item: item.score, reverse=True)
+    pool = sortedData[0: poolLength]
+
+    data = []
+    while len(data) < population:
+        parentA = pool[random.randint(0,len(pool)-1)]
+        parentB = pool[random.randint(0,len(pool)-1)]
+
+        parentAScore = parentA.score / (parentA.score + parentB.score)
```
Comment (QuantumNovice) [inline other/genetic_algorithm.py]: Handle `ZeroDivisionError` either as 
`parentAScore = parentA.score / (1 + parentA.score + parentB.score)`
or set initial greater than zero.

File: other/genetic_algorithm.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        self.target = target
+        self.data = data
+        self.score = self.get_score()
+
+    def get_score(self):
+        score = 0
+        for i in range(len(self.data)):
+            if self.data[i] == self.target[i]:
+                score += 1
+        return score // len(self.data)
```
Comment (QuantumNovice) [inline other/genetic_algorithm.py]: Change `int division` to `float division` as the code gets stuck in infinite loop. Therefore no improvement is made in new generations.
`return score / len(self.data)`

File: other/genetic_algorithm.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import random
+
+population = 200
+poolLength = 20
+generations = 0
+mutation = 0.01
+
+alphabet = "abcdefghijklmnopqrstuvwxyz! "
```
Comment (QuantumNovice) [inline other/genetic_algorithm.py]: This is not necessary but it's always better.
```python
import string
alphabets = string.ascii_lowercase +'! '
```

**Pull request closed without merging**